packages/go_router/test/go_router_test.dart

```diff
@@ -187,7 +187,8 @@ void main() {
     testWidgets('match top level route when location has trailing / (2)',
         (WidgetTester tester) async {
       final List<GoRoute> routes = <GoRoute>[
-        GoRoute(path: '/profile', redirect: (_) => '/profile/foo'),
+        GoRoute(
+            path: '/profile', builder: dummy, redirect: (_) => '/profile/foo'),
```

Comment:
This seems like a breaking change? makes builder optional seems to be fine if user just want to redirect a specific route

Reply:
Since we can't guarantee that `redirect` won't return null we need to make builder required

Reply:
We need to bump the major version if we add breaking changes. Can we punt this breaking change until later?

Reply:
We can throw if it does point to route without builder.

Reply:
Yeah I can change this back

Reply:
Actually, I'm having trouble changing this back without making the tests in go_router_test.dart fail. I'm going to leave this as-is.

Reply:
We have this test:

```dart
  test('throws when a builder is not set', () {
    expect(() => GoRoute(path: '/'), throwsA(isAssertionError));
  });
```